eslzArm/managementGroupTemplates/policyAssignments/AUDIT-AlzCorpNetworkingPolicyAssignment.json

```diff
@@ -0,0 +1,57 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-08-01/managementGroupDeploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "policyEffect": {
+            "type": "string",
+            "allowedValues": [
+                "Deny",
+                "Audit"
+            ],
+            "defaultValue": "Audit"
+        },
+        "enforcementMode": {
+            "type": "string",
+            "allowedValues": [
+                "Default",
+                "DoNotEnforce"
+            ],
+            "defaultValue": "Default"
+        }
+    },
+    "variables": {
+        "policyDefinitions": {
+            "denyCorpNetworking": "/providers/Microsoft.Authorization/policyDefinitions/6c112d4e-5bc7-47ae-a041-ea2d9dccd749"
+        },
+        "policyAssignmentNames": {
+            "denyCorpNetworking": "Audit-AlzCorp-Networking",
+            "description": "Audits the deployment of Private Link / Private DNS Zone resources in the Corp landing zone.",
+            "displayName": "Audit Private Link / Private DNS Zone resources"
+        }
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Authorization/policyAssignments",
+            "apiVersion": "2019-09-01",
+            "name": "[variables('policyAssignmentNames').denyCorpNetworking]",
+            "properties": {
+                "description": "[variables('policyAssignmentNames').description]",
+                "displayName": "[variables('policyAssignmentNames').displayName]",
+                "policyDefinitionId": "[variables('policyDefinitions').denyCorpNetworking]",
+                "enforcementMode": "[parameters('enforcementMode')]",
+                "parameters": {
+                    "listOfResourceTypesNotAllowed": {
+                        "value": [
+                            "microsoft.network/privatednszones",
+                            "microsoft.network/privatelinkservices"
```

Comment:
```suggestion
```
Remove this as we dont want to block the use of Private Link, just the Private DNS zones for private link, as documented here: https://learn.microsoft.com/en-us/azure/private-link/private-endpoint-dns#azure-services-dns-zone-configuration

Think this policy assignment may need some re-thinking. The logic will need to be, if type == microsoft.network/privatednszones and the names of the zones are "in" the array of DNS zones from https://learn.microsoft.com/en-us/azure/private-link/private-endpoint-dns#azure-services-dns-zone-configuration then deny/audit.

But we do not want to block private dns zones or private link all up

Reply:
Understood and dropped Private Link from the scope, however, not this is an AUDIT policy, we're not currently blocking anything.

Reply:
@Springstone as this was in our agreed list of things to do, we should do it. And yes good spot on it's audit, i jumped the gun there.

Here is a policy that will do what we want it to do, that i just made and tested:

```json
{
  "policyType": "Custom",
  "mode": "Indexed",
  "displayName": "Audit the creation of Private Link Private DNS Zones",
  "description": "This policy audits the creation of a Private Link Private DNS Zones in the current scope, used in combination with policies that create centralized private DNS in connectivity subscription",
  "metadata": {
    "version": "1.0.0",
    "category": "Network",
    "source": "https://github.com/Azure/Enterprise-Scale/",
    "alzCloudEnvironments": [
      "AzureCloud",
      "AzureChinaCloud",
      "AzureUSGovernment"
    ]
  },
  "parameters": {
    "effect": {
      "type": "String",
      "allowedValues": [
        "Audit",
        "Deny",
        "Disabled"
      ],
      "defaultValue": "Audit",
      "metadata": {
        "displayName": "Effect",
        "description": "Enable or disable the execution of the policy"
      }
    },
    "privateLinkDnsZones": {
      "type": "Array",
      "metadata": {
        "displayName": "Private Link Private DNS Zones",
        "description": "An array of Private Link Private DNS Zones to check for the existence of in the assigned scope."
      },
      "defaultValue": [
        "privatelink.azure-automation.net",
        "privatelink.database.windows.net"
      ]
    }
  },
  "policyRule": {
    "if": {
      "allOf": [
        {
          "field": "type",
          "equals": "Microsoft.Network/privateDnsZones"
        },
        {
          "field": "name",
          "in": "[parameters('privateLinkDnsZones')]"
        }
      ]
    },
    "then": {
      "effect": "[parameters('effect')]"
    }
  }
}
```

And for ease you can default the values in the array to the same list as here https://github.com/Azure/Enterprise-Scale/blob/main/eslzArm/eslzArm.json#L871-L937

And then actually during the assignment, if you pass into the parameter from the `eslzArm.json` ARM template the variable called `privateDnsZonesMerge` https://github.com/Azure/Enterprise-Scale/blob/main/eslzArm/eslzArm.json#L1008 then this will put all the zones in based on the region the customer deploys to. It will be slick. 

So lets add this new custom policy and assign it please as we initially agreed to do it in the spreadsheet.